### PR TITLE
Issue 1438/checked state

### PR DESCRIPTION
**Pull request by stephanegigandet**
Added a way for experimented contributors to mark products as checked.

### CODE REVIEW

**Comment by teolemon** [comment]
```Testing po/tags/en.po .po fileUse of uninitialized value in concatenation (.) or string at scripts/check_po_file.pl line 58.
Error: empty msgstr string for msgctxt last_check_dates:plural - msgid 
1 errors```

File: lib/ProductOpener/Display.pm [hunk 1]
```diff
@@ -7143,11 +7143,19 @@ HTML
 		$other_editors = "<br>\n$Lang{also_edited_by}{$lang} ${other_editors}.";
 	}
 
+	my $checked = "";
+	if ((defined $product_ref->{checked}) and ($product_ref->{checked} eq 'on')) {
+		my $last_checked_date = display_date_tag($product_ref->{last_checked_t});
+		my $last_checker = "<a href=\"" . canonicalize_tag_link("users", get_fileid($product_ref->{last_checker})) . "\">" . $product_ref->{last_checker} . "</a>";
+		$checked = "<br/>\n$Lang{product_last_checked}{$lang} $last_checked_date $Lang{by}{$lang} $last_checker.";
```
Comment (hangy) [inline lib/ProductOpener/Display.pm]: Should we maybe change this translation to use sprintf? The sentence structure may be different in other languages than the current "Last check of product page on" folowed by the three values. @teolemon

File: cgi/product_multilingual.pl [hunk 2]
```diff
@@ -615,6 +615,34 @@ ($$)
 		
 	}
 	
+	
+	# product check
+	
+	if ($admin) {
```
Comment (hangy) [inline cgi/product_multilingual.pl]: What do you think about splitting this up to a new user group? Being able to delete products is potentially much more destructive than the ability to set the checked state.

File: po/common/en.po [hunk 3]
```diff
@@ -2968,3 +2968,36 @@ msgctxt "additives_classes"
 msgid "Functions"
 msgstr "Functions"
 
+msgctxt "photos_and_data_check"
+msgid "Photos and data check"
+msgstr "Photos and data check"
+
+msgctxt "photos_and_data_check_description"
+msgid "Product pages can be marked as checked by experimented contributors who verify that the most recent photos are selected and cropped, and that all the product data that can be inferred from the product photos has been filled and is correct."
```
Comment (hangy) [inline po/common/en.po]:
```suggestion
msgid "Product pages can be marked as checked by experienced contributors who verify that the most recent photos are selected and cropped, and that all the product data that can be inferred from the product photos has been filled and is correct."
```

File: po/common/en.po [hunk 4]
```diff
@@ -2968,3 +2968,36 @@ msgctxt "additives_classes"
...
+msgstr "Product pages can be marked as checked by experimented contributors who verify that the most recent photos are selected and cropped, and that all the product data that can be inferred from the product photos has been filled and is correct."
```
Comment (hangy) [inline po/common/en.po]:
```suggestion
msgstr "Product pages can be marked as checked by experienced contributors who verify that the most recent photos are selected and cropped, and that all the product data that can be inferred from the product photos has been filled and is correct."
```

File: po/common/fr.po [hunk 5]
```diff
@@ -2957,3 +2957,35 @@ msgctxt "additives_classes"
 msgid "Functions"
 msgstr "Fonctions"
 
+msgctxt "photos_and_data_check"
+msgid "Photos and data check"
+msgstr "Vérification des photos et des données"
+
+msgctxt "photos_and_data_check_description"
+msgid "Product pages can be marked as checked by experimented contributors who verify that the most recent photos are selected and cropped, and that all the product data that can be inferred from the product photos has been filled and is correct."
```
Comment (hangy) [inline po/common/fr.po]:
```suggestion
msgid "Product pages can be marked as checked by experienced contributors who verify that the most recent photos are selected and cropped, and that all the product data that can be inferred from the product photos has been filled and is correct."
```

File: po/common/common.pot [hunk 6]
```diff
@@ -3141,3 +3141,35 @@ msgctxt "additives_classes"
 msgid "Functions"
 msgstr ""
 
+msgctxt "photos_and_data_check"
+msgid "Photos and data check"
+msgstr ""
+
+msgctxt "photos_and_data_check_description"
+msgid "Product pages can be marked as checked by experimented contributors who verify that the most recent photos are selected and cropped, and that all the product data that can be inferred from the product photos has been filled and is correct."
```
Comment (hangy) [inline po/common/common.pot]:
```suggestion
msgid "Product pages can be marked as checked by experienced contributors who verify that the most recent photos are selected and cropped, and that all the product data that can be inferred from the product photos has been filled and is correct."
```

File: cgi/product_multilingual.pl (hunk 2 above)
Comment (stephanegigandet) [reply cgi/product_multilingual.pl]: That's right, we can create a moderators group.

File: lib/ProductOpener/Display.pm (hunk 1 above)
Comment (stephanegigandet) [reply lib/ProductOpener/Display.pm]: For this one I just copied what we had for the last_edit_date. It's a good point. I'm not sure if sprintf is the good solution, the %s might be confusing for translators. Maybe we could use <tags> instead. e.g "Last verification by <user> on <date>", that way the word order could be changed if needed.

This is something we should probably discuss separately and apply across all the strings. cc @teolemon

**Pull request merged**